src/Frontend/state/variables/use-filtered-attributions.ts

```diff
@@ -0,0 +1,56 @@
+// SPDX-FileCopyrightText: Meta Platforms, Inc. and its affiliates
+// SPDX-FileCopyrightText: TNG Technology Consulting GmbH <https://www.tngtech.com>
+//
+// SPDX-License-Identifier: Apache-2.0
+import { useEffect } from 'react';
+
+import { Attributions } from '../../../shared/shared-types';
+import {
+  Filter,
+  FilterCounts,
+} from '../../web-workers/scripts/get-filtered-attributions';
+import { useAppSelector } from '../hooks';
+import { getManualAttributions } from '../selectors/all-views-resource-selectors';
+import { useVariable } from './use-variable';
+
+export const FILTERED_ATTRIBUTIONS = 'filtered-attributions';
+
+export interface FilteredAttributions {
+  selectedFilters: Array<Filter>;
+  attributions: Attributions;
+  counts: FilterCounts | null;
+  loading: boolean;
+  search: string;
+}
+
+export const initialFilteredAttributions: FilteredAttributions = {
+  selectedFilters: [],
+  attributions: {},
+  counts: null,
+  loading: false,
+  search: '',
+};
+
+export function useFilteredAttributions() {
+  const attributions = useAppSelector(getManualAttributions);
+  const [filteredAttributions, setFilteredAttributions] =
+    useVariable<FilteredAttributions>(
+      FILTERED_ATTRIBUTIONS,
+      initialFilteredAttributions,
+    );
+
+  useEffect(() => {
+    if (!filteredAttributions.selectedFilters.length) {
+      setFilteredAttributions((prev) => ({
+        ...prev,
+        attributions,
+      }));
```

Comment:
Why do we pass an arrow function here instead of just passing `attributions`?
```suggestion
      setFilteredAttributions(
        attributions,
      );
```

Reply:
because i need to preserve the other attributes that are inside `prev` but not part of `attributions`